src/Chapter2.hs

```diff
@@ -336,7 +336,9 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList x y l
+  | x > y || x > length l || x < 0 || y < 0 = []
```

Comment:
In lists, to get `length` is very expensive process. You need to walk each element to count that, so I would not advice to have this check in here 🙂 You actually don't need to check `x > length l` at all, as `take` and `drop` handle these cases okay 👍🏼